src/components/costindex/CostIndexTable.tsx

```diff
@@ -0,0 +1,236 @@
+import React, { useState } from 'react';
+import moment from 'moment/moment';
+import { get, SubmitHandler, useForm } from 'react-hook-form';
+import { useTranslation } from 'react-i18next';
+import {
+  Button,
+  DateInput,
+  Dialog,
+  IconPlus,
+  IconQuestionCircle,
+  LoadingSpinner,
+  Notification,
+  Table,
+  TextInput,
+} from 'hds-react';
+import * as yup from 'yup';
+import { yupResolver } from '@hookform/resolvers/yup/dist/yup';
+
+import { getCurrentLangCode } from '../../utils/getCurrentLangCode';
+import { useAddCostIndexMutation, useGetCostIndexesQuery } from '../../redux/services/api';
+import Container from '../common/container/Container';
+import { AddEditCostIndex } from '../../types';
+import parseApiErrors from '../../utils/parseApiErrors';
+import { toast } from '../common/toast/ToastManager';
+import { usePageTitle } from '../../utils/usePageTitle';
+import CostIndexSingleTable from './CostIndexSingleTable';
+
+import styles from './CostIndexTable.module.scss';
+
+const T_PATH = 'components.costindex.CostIndexTable';
+
+const CostIndexTable = (): JSX.Element => {
+  const { t: translate } = useTranslation();
+  const t = (label: string) => translate(`${T_PATH}.${label}`);
+  usePageTitle(t('indexValue'));
+
+  const { data, isLoading, isError } = useGetCostIndexesQuery();
+  const [addCostIndex, { isLoading: addIsLoading }] = useAddCostIndexMutation();
+
+  const confirmationDialogTarget = React.useRef(null);
+  const openConfirmationDialogButtonRef = React.useRef(null);
+  const [isOpen, setIsOpen] = React.useState(false);
+  const [errorMessages, setErrorMessages] = useState<string[]>([]);
+  const close = () => setIsOpen(false);
+  const titleId = 'info-dialog-title';
+  const descriptionId = 'info-dialog-content';
+
+  const schema = yup.object({
+    valid_from: yup
+      .string()
+      .trim()
+      .matches(/\d{1,2}\.\d{1,2}\.\d{4}/, t('invalidDate'))
+      .required(t(`required`)),
+    value: yup
+      .string()
+      .trim()
+      .matches(/^\d+([.,]\d{1,2})?$/, t('invalidValue'))
+      .required(t(`required`)),
+  });
+  const { handleSubmit, register, setValue, formState, watch, reset, trigger, getValues } = useForm<AddEditCostIndex>({
+    resolver: yupResolver(schema),
+  });
+
+  const errors = formState.errors;
+  const validFromDate = watch('valid_from');
+  const formId = `cost-index-add-form`;
+
+  const validFrom = getValues('valid_from');
+  const value = getValues('value');
+
+  if (isLoading) {
+    return <LoadingSpinner small />;
```

Comment:
Could you change this component to the local `<Spinner />`?

I noticed this small spinner flash outside the container, just at the left screen edge.